lib/utils/scope-subtree.js

```diff
@@ -0,0 +1,76 @@
+/**
+@license
+Copyright (c) 2019 The Polymer Project Authors. All rights reserved.
+This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+*/
+
+import './boot.js';
+import { wrap } from './wrap.js';
+
+const ShadyDOM = window.ShadyDOM;
+const ShadyCSS = window.ShadyCSS;
+
+/**
+ * Ensure that elements in a ShadowDOM container are scoped correctly.
+ *
+ * @param  {!Element} container Container element to scope
+ * @param  {boolean=} shouldObserve if true, start a mutation observer for added nodes to the container
+ * @return {?MutationObserver} Returns a new MutationObserver on `container` if `shouldObserve` is true.
+ */
+export function scopeSubtree(container, shouldObserve = false) {
+  // If using native ShadowDOM, abort
+  if (!ShadyDOM || !ShadyCSS) {
+    return null;
+  }
+  // ShadyCSS handles DOM mutations in patched mode
+  if (!ShadyDOM['noPatch']) {
+    return null;
+  }
+  const ScopingShim = ShadyCSS['ScopingShim'];
+  // if ScopingShim is not available, abort
+  if (!ScopingShim) {
+    return null;
+  }
+  // capture correct scope for container
+  const containerScope = ScopingShim['scopeForNode'](container);
+
+  const scopify = (node) => {
+    const wrappedNode = wrap(node);
```

Comment:
To match Polymer 1 behavior, we should use native querySelector: `ShadyDOM.nativeMethods.querySelectorAll.call(node, ...)`